spec/facility_factory_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'spec_helper'
+
+RSpec.describe FacilityFactory do
+  before (:each) do 
+    # @factory = FacilityFactory.new
+    @Colorado = DmvDataService.new.co_dmv_office_locations
+    @new_york_facilities = DmvDataService.new.ny_dmv_office_locations
+    @missouri_facilities = DmvDataService.new.mo_dmv_office_locations
+  end
+  describe '#Facility Factory' do
+    it 'shows us Colorado data' do
+      factory = FacilityFactory.new
+      expect(factory).to be_a(FacilityFactory)
+      facilities = factory.create_facility(@Colorado)
+      expect(facilities).to be_a(Array)
+      expect(@Colorado[0][:dmv_office]).to be_a(String)
+      expect(facilities[0]).to be_a(Facility)
+      expect(@Colorado[0][:address_li]).to be_a(String)
+      expect(@Colorado[0][:address__1]).to be_a(String)
+      expect(@Colorado[0][:city]).to be_a(String)
+      expect(@Colorado[0][:state]).to be_a(String)
+      expect(@Colorado[0][:state]).to eq("CO")
```

Comment:
Right here, you're testing the data from teh api which isn't really necessary. We mainly need you to test the data that's getting returned to you from your `create_facility` method.

Reply:
It looks like your return value for that method is stored in `facilities`. You could do several things. For starters, I'd suggest just pulling out the first element in that array (`facilities.first`) and making sure it's a Facility object, and then making sure that each of it's expected attributes are not nil (that woul dmean that your attribute assignments were going well).